Data/Aeson/Parser/Internal.hs

```diff
@@ -323,37 +316,20 @@ jstring = A.word8 DOUBLE_QUOTE *> jstring_
 -- | Parse a string without a leading quote.
 jstring_ :: Parser Text
 {-# INLINE jstring_ #-}
-jstring_ = {-# SCC "jstring_" #-} do
-#if MIN_VERSION_ghc_prim(0,3,1)
-  (s, S _ escaped) <- A.runScanner startState go <* A.anyWord8
-  -- We escape only if there are
-  -- non-ascii (over 7bit) characters or backslash present.
-  --
-  -- Note: if/when text will have fast ascii -> text conversion
-  -- (e.g. uses utf8 encoding) we can have further speedup.
-  if isTrue# escaped
-    then case unescapeText s of
-      Right r  -> return r
-      Left err -> fail $ show err
-    else return (TE.decodeUtf8 s)
- where
-    startState              = S 0# 0#
-    go (S skip escaped) (W8# c)
-      | isTrue# skip        = Just (S 0# escaped')
-      | isTrue# (w ==# 34#) = Nothing   -- double quote
-      | otherwise           = Just (S skip' escaped')
-      where
-        w = word2Int# c
-        skip' = w ==# 92# -- backslash
-        escaped' = escaped
-            `orI#` (w `andI#` 0x80# ==# 0x80#) -- c >= 0x80
-            `orI#` skip'
-            `orI#` (w `andI#` 0x1f# ==# w)     -- c < 0x20
-
-data S = S Int# Int#
-#else
+jstring_ = do
+  s <- A.takeWhile (\w -> w /= DOUBLE_QUOTE && w /= BACKSLASH && not (testBit w 7))
```

Comment:
Is the `testBit w 7` check to see whether we are only dealing with ASCII-characters in the UTF-8 encoding?

To my great dismay, the JSON RFC states
> JSON text SHALL be encoded in UTF-8, UTF-16, or UTF-32.

but non-utf8 is way out of scope for aeson anyway I assume.

Reply:
RFC8259 https://tools.ietf.org/html/rfc8259 says 

>   JSON text exchanged between systems that are not part of a closed
   ecosystem MUST be encoded using UTF-8 [RFC3629].

It also `Obsoletes: 7159`.

But RFC7159 Indeed talked about other encodings, see https://tools.ietf.org/html/rfc7159

Reply:
Ah, great to know.

But does `testBit w 7` do the ASCII check?

Reply:
It's rather checks for UTF-8 characters that take more then 1 byte. IIRC it's necessary only because multibyte code point can have subsequent bytes equal to double quote or backslash, and we don't want to confuse that bytes with actual quote and backslash.

Reply:
I actually don't remember why the upper bound check is there. You definitely need one if you would like to use `decodeLatin1` which might be slightly faster than `decodeUtf8` - whether that matters with current `text` - I don't know.

If none tests fails if it's removed, then it indeed might no be necessary. :)